src/CoreBundle/Service/PortalDomainServiceInterface.php

```diff
@@ -86,11 +86,9 @@ public function hasPrimaryDomain($portalId = null, $languageId = null);
     /**
      * Sets the currently active portal.
      *
-     * @param TCMSPortal $portal
-     *
      * @return void
      */
-    public function setActivePortal(TCMSPortal $portal);
+    public function setActivePortal(TCMSPortal $portal = null);
```

Comment:
If you allow null, then the method should be able to handle a call with null. So the event should take null as a new portal.

Reply:
i changed the event to handle null values